### ISSUE TITLE
Distribute iD as a mobile app

### ISSUE THREAD

**Issue by quincylvania** [openstreetmap/iD#7639]
This idea may sound rather ambitious, but hear me out 😄

Over the past few months I've been putting in a lot of work toward achieving parity between mouse and touch interaction methods in iD. For a taste, see these closed issues: #5505, #7423, #6049, #2128, #2677, #7415, #6598, #7380, #7396, #7577, #7432, #1981, #7590.

The results are quite promising. I've done a few medium-sized changesets with iD on iPad without too many compromises, and the experience will only get better as we continue to commit to touch as a first-class workflow. Still, no matter how excellent an experience we build, using a mobile browser page will never be as nice as using a "real" app.

The solution? We can wrap iD in a web view and distribute it for free on the various app stores. Mobile users are far more accustomed to downloading apps than navigating websites. As such, this could be a terrific opportunity to grow the OSM community.

A few mobile OSM editors do already exist, each with their own unique character. iD would simply be another option and not aim to replace these.

A number of technical and administrative details remain to be worked out before we can run with this idea. For instance, iD does not scale well to tiny screens even if it handles touch input okay. We'd also have to determine what the app is called and who the named publisher should be on the store pages.

This is probably still a long way off, but I wanted to get the discussion started 📱🚀

**Comment by manfredbrandl** [comment]
I propose that the OSM Foundation should publish the APP. A possible Name may be „OSM Editor iD“.

**Comment by DerKarlos** [comment]
I am exited. When iD was started, my first expectation was iD to be usable on my new iPad then. But as I asked for this feature, that topic was not on the list :-/
Now it starts and I would like to help, to test, to experiment with different solutions.
[Try](http://ac1000.de/tosme/) my [first steps](https://wiki.openstreetmap.org/wiki/TosmE) to edit on iPad. (Don't test with mouse, there are offsets optimized for touching righthand fingers :-)

You may consider to make it a Progressive-Web-App (PWA) , usable for all devices. Ok, in iOS it needs to be wrapped in an native app this days yet.
iD is a plain Javascript solution now, is'nt it? If you face slow download or reactions or memory limits on older devices, you may consider to use WASM too.

**Comment by eneerhut** [comment]
I'm a huge fan of iD @quincylvania and how it maintains a balance between ease of use and feature richness. My question with the app would be what problems/gaps it aims to fill.

My experience with existing OSM mobile apps is as follows:
- **[OSMAnd](https://play.google.com/store/apps/details?id=net.osmand&hl=en_US)** (cross platform): Decent for adding points of interest. I've only used in on Android. I believe the iOS version is behind on features.
- **[MAPS.ME](https://play.google.com/store/apps/details?id=com.mapswithme.maps.pro&hl=en_US)** (cross platform): Also decent for adding points of interest.
- **[Vespucci](https://play.google.com/store/apps/details?id=de.blau.android&hl=en_US)** (Android): Fully featured mobile editing. Incredibly capable with a lot of thought behind it. Not an app I get out on the move often because it always feels a bit involved for simple edits.
- **[StreetComplete](https://play.google.com/store/apps/details?id=de.westnordost.streetcomplete&hl=en_US)** (Android): Quest focused mobile app. I absolutely love this app and the innovative approach it has brought to mobile editing. It allows users to answer simple questions about the world around them to help improve the map. The UI is fantastic and continually improves. I only wish there were even more quests on offer in each area.
- **[Go Map!!](https://apps.apple.com/us/app/go-map/id592990211)** (iOS): Very capable mobile app. Useful for both simple edits like points of interest and slightly more complex edits like fairways on a golf course. 
- **[Observe](https://github.com/developmentseed/observe)** (cross platform): Still in beta, but differs from the others in its field centric, offline first approach. I haven't spent too much time with it, and felt the lack of tag presets limiting, but I'm optimistic about this one.

I'm grateful to all the developers contributing to the above. With the exception of Observe which is still alpha, I've been using them all and they each have a role to play. For my own personal use case, I see Go Map!! becoming the one I will use regularly. It allows me to do most of the editing I would want to do on the move in a fairly quick, intuitive manner.

Perhaps a mobile version of iD editor would achieve similar, but with the benefit of being cross platform and incorporating some of the best aspects of the web experience.

**Comment by eneerhut** [comment]
An additional thought to the above. I have run quite a few mapathons where people attend with their tablets and want to follow along in the same way those with their laptops are mapping.

They can get iD editor to work with a bit of effort, but they don't have the same experience as those on their laptop. As you mentioned, a mobile iD editor could be a helpful introduction to OSM for the less tech savvy and/or lower income communities who may not have access to a full features laptop.

Of course quality control needs to be front of mind and perhaps an extended sandbox/tutorial mode would be required in these scenarios.

**Comment by DerKarlos** [comment]
I agree with @eneerhut, tablets editing on the field eye to eye tho the ground truth is great.
I also like "go map!!", we should consider to take some ideas of its UI.

To me, a main advantage of iD is: Just start it and edit, no installation. And mostly selves explaining too.
So I hope, this will be the same on tablets.
Could it be offline to? May be not at first, but technically, it should be possible.

(Didn't know "observe" but ill ask for a testflight)

**Comment by quincylvania** [comment]
> Now it starts and I would like to help, to test, to experiment with different solutions.
> Try my first steps to edit on iPad. 

@DerKarlos Interesting! We've already solved the touch drawing interaction in iD, but there's always room for more editors if you're going to continue building that one.

> iD is a plain Javascript solution now, is'nt it? If you face slow download or reactions or memory limits on older devices, you may consider to use WASM too.

Unfortunately, we don't have the development resources to even consider a full or partial rewrite of iD in a different language. But there's been some talk of using canvas to speed up map rendering (#3741).

> Could it be offline to? May be not at first, but technically, it should be possible.

Possibly one day, but it'd be a big shift. It's not an initial priority.

---

@eneerhut Thanks for the overview of the current mobile editor landscape! It's important to have a full picture of the space.

> My question with the app would be what problems/gaps it aims to fill.

My goal would be for iD on all major mobile platforms to have full feature parity with desktop iD. Thus, users could expect a consistent mapping experience, full suite of editing tools, and active development support no matter what device they use for editing. Though ultimately each mobile editor has its own strengths and weaknesses, so mappers can chose the one that best suits their needs. Like I said, iD would simply be another option.

> a mobile iD editor could be a helpful introduction to OSM for the less tech savvy and/or lower income communities who may not have access to a full features laptop.

Yes!

> Of course quality control needs to be front of mind and perhaps an extended sandbox/tutorial mode would be required in these scenarios.

Yep, we'd only ship a quality mapping experience. And we do need to update the walkthrough for touch interaction.

**Comment by mcorteel** [comment]
I use iD on my phone regularly and I love it! The two pain points that I see are:
- performance
- running it in a browser.

A progressive web app would probably be the easiest way to solve the second point. Though most features seem very complicated to have offline, they could be added in time. In the meantime, it would fail more gracefully and feel more integrated. And it would require no extra infrastructure.

For performance, the PWA might help a little, but there's not much to do: better hardware = better performance. I want to point out that on my 2017 Pixel 2, the performance is pretty good, though some lags make me move points when I just want to scroll the view (other than that, it works great). I don't own a tablet, so I can't compare.

TLDR; a PWA would be awesome 🙂

**Comment by danieldegroot2** [comment]
See also https://github.com/openstreetmap/iD/issues/10244#issuecomment-2455138330 (a demo site which is also a pwa)
